go-controller/pkg/libovsdbops/template_var.go

```diff
@@ -0,0 +1,109 @@
+package libovsdbops
+
+import (
+	"context"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	libovsdb "github.com/ovn-org/libovsdb/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+)
+
+// ListTemplates looks up all chassis template variables.
+func ListTemplates(nbClient libovsdbclient.Client) ([]*nbdb.ChassisTemplateVar, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), types.OVSDBTimeout)
+	defer cancel()
+
+	templatesList := []*nbdb.ChassisTemplateVar{}
+	err := nbClient.List(ctx, &templatesList)
+	if err != nil {
+		return nil, err
+	}
+	return templatesList, nil
+}
+
+func mutateChassisTemplateVarOps(nbClient libovsdbclient.Client, preserveVarValues bool,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	modelClient := newModelClient(nbClient)
+	return modelClient.CreateOrUpdateOps(ops, []operationModel{
+		{
+			Model: template,
+			ModelPredicate: func(item *nbdb.ChassisTemplateVar) bool {
+				return item.Chassis == template.Chassis
+			},
+			OnModelMutations:                  []interface{}{&template.Variables},
+			ErrNotFound:                       false,
+			BulkOp:                            false,
+			PreserveExistingMapValuesOnMutate: preserveVarValues,
+		},
+	}...)
+}
+
+func CreateChassisTemplateVarOps(nbClient libovsdbclient.Client,
+	ops []libovsdb.Operation, template *nbdb.ChassisTemplateVar) ([]libovsdb.Operation, error) {
+
+	return mutateChassisTemplateVarOps(nbClient, true, ops, template)
```

Comment:
If you put it like that it makes sense.  

The only reason to have `PreserveExistingMapValuesOnMutate=true` is for keys that did not exist.  For these we could try to avoid the extra cost on the server side.  But OK, this is very theoretic and I don't think will impact performance that much.  Especially with the "shared" backends optimization we only end up with very few template variables in the database.

The source of truth is ovn-kubernetes so it should always reconcile whatever ends up in the DB with whatever state it has in memory.  I'll change it like that.

Reply:
I removed the commit that adds `PreserveExistingMapValuesOnMutate` and I fixed the behavior to `PreserveExistingMapValuesOnMutate=false` by default (non-configurable).